### PR TITLE
feat: add init function support

### PR DESCRIPTION
**Pull request by exaby73**

### CODE REVIEW

File: src/firebase_functions/core.py [hunk 1]
```diff
@@ -80,3 +82,48 @@ class Change(_typing.Generic[T]):
...
+    global _init_callback
+
+    _init_callback = callback
+
+    if _did_init:
+        _logger.warn(
+            "Setting init callback more than once. Only the most recent callback will be called"
+        )
+
+    _init_callback = callback
```
Comment (inlined) [inline src/firebase_functions/core.py:108]: redundant

File: tests/test_identity_fn.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+                    method="POST",
+                    json={
+                        "data": {
+                            "jwt": "jwt"
+                        },
+                    },
+                ).get_environ()
+                request = Request(environ)
+                decorated_func = identity_fn.before_user_signed_in()(func)
+                print(decorated_func(request))
```
Comment (inlined) [inline tests/test_identity_fn.py:62]: Shouldn't print, right?

**Review by inlined — APPROVED** [review]
Few nits, but looks good to me

File: src/firebase_functions/core.py (hunk 1 above)
Comment (exaby73) [reply src/firebase_functions/core.py:108]: Oops. Removed it

**Pull request merged**